extensions/wikia/Email/Controller/WallMessageController.class.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\EmailController;
+
+class WallMessageController extends EmailController {
+
+	/**
+	 * @template wallMessage
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->request->getVal( 'fromName' ),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'wallMessageTitle' => $this->request->getVal( 'messageTitle' ),
```

Comment:
We've been creating new instance variables for any controller parameters that aren't already known by the parent controller in an ```initEmail``` method.  This makes it easy to see what a controller intends on using and provides an opportunity to do error checking before starting.